connect/runtime/src/test/java/org/apache/kafka/connect/runtime/health/ConnectClusterStateImplTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.runtime.health;
+
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.runtime.Herder;
+import org.apache.kafka.connect.runtime.HerderProvider;
+import org.apache.kafka.connect.util.Callback;
+import org.easymock.EasyMock;
+import org.easymock.IAnswer;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+
+public class ConnectClusterStateImplTest {
+  
+    protected static final long CONNECTORS_TIMEOUT_MS = 2_000;
+    
+    protected Herder mockHerder;
+    protected HerderProvider herderProvider;
+    protected ConnectClusterStateImpl connectClusterState;
+    protected Collection<String> expectedConnectors;
+    
+    @Before
+    public void setUp() {
+        mockHerder = EasyMock.mock(Herder.class);
+        herderProvider = new HerderProvider(mockHerder);
+        connectClusterState = new ConnectClusterStateImpl(herderProvider, CONNECTORS_TIMEOUT_MS);
+        expectedConnectors = Arrays.asList("sink1", "source1", "source2");
+    }
+    
+    @Test
+    public void connectors() {
```

Comment:
Thanks for simplifying the tests @C0urante! Definitely on the right track. 
As you can see in https://github.com/apache/kafka/blob/trunk/connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaConfigBackingStoreTest.java 
you can use the already existing `TestFuture` and be a bit more specific on what to capture and what you expect. 

For example this could be written as: 
```java
    @Test
    public void connectors() {
        Capture<Callback<Collection<String>>> callback = EasyMock.newCapture();
        mockHerder.connectors(EasyMock.capture(callback));
        EasyMock.expectLastCall().andAnswer(new IAnswer<Future<Collection<String>>>() {
            @Override
            public Future<Collection<String>> answer() throws Throwable {
                TestFuture<Collection<String>> future = new TestFuture<>();
                callback.getValue().onCompletion(null, expectedConnectors);
                future.resolveOnGet(expectedConnectors);
                return future;
            }
        });
        EasyMock.replay(mockHerder);
        assertEquals(expectedConnectors, connectClusterState.connectors());
    }
```
this way you also manage to avoid unchecked casts.

Reply:
@kkonstantine definitely prefer the `Capture`-based approach that you've outlined, but I don't think the `TestFuture`/`IAnswer` combination in your example is applicable since the future/callback we're dealing with is being passed to `Herder.connectors(...)`, instead of being returned by it.

Reply:
That's correct. It's called in the method you test. I was carried away in the effort to remove the unchecked casts.